### PR TITLE
Add vault-configs, weekly pipeline yaml and re enable Secret Manager stages in the CI pipeline

### PR DESCRIPTION
**Pull request by dkurepa**
https://github.com/dotnet/arcade-services/issues/2316
This PR adds the weekly pipeline yaml and the .vault configs that it needs.
Only secrets that are used by the pipeline have been transferred.
List of secrets that were **not** transferred:

Helixkv.yaml
- dnb2-bot-account-redmond
- dn-dependabot-account-redmond
- dotnet-mc-bot-account

maestro-secrets.yaml
- data-protection-key-file-uri
- maestro-memory-dumps-container-sas-uri
- prod-maestro-token
- dn-bot-dnceng-build-rw-code-rw-release-rw
- dn-bot-devdiv-build-rw-code-rw-release-rw
- dn-bot-domoreexp-build-rw-code-rw-release-rw
- dn-bot-dnceng-packaging-rwm

maestroint.yaml
- build-asset-registry-admin-connection-string
- build-asset-registry-sql-connection-string

maestrolocal.yaml
- prod-maestro-token
- dn-bot-dnceng-build-rw-code-rw-release-rw
- dn-bot-devdiv-build-rw-code-rw-release-rw
- dn-bot-domoreexp-build-rw-code-rw-release-rw
- dn-bot-dnceng-packaging-rwm

maestroprod.yaml
- build-asset-registry-admin-connection-string
- build-asset-registry-sql-connection-string

### CODE REVIEW

**Comment by dkurepa** [comment]
We had to add the maestro keyvaults configuration because of the telemetry service, which feels wrong. Created this issue as a follow up to fix this https://github.com/dotnet/arcade/issues/13608

**Comment by garath** [comment]
Secret-manager has a feature that writes a warning to the console if it finds a secret in a vault but not in the manifest. Since this is the first real case of intentionally splitting secrets for a vault across multiple manifests, consider also removing this secret manager. These warnings will now just be "expected" but cause a lot of noise.

**Comment by dkurepa** [comment]
> Secret-manager has a feature that writes a warning to the console if it finds a secret in a vault but not in the manifest. Since this is the first real case of intentionally splitting secrets for a vault across multiple manifests, consider also removing this secret manager. These warnings will now just be "expected" but cause a lot of noise.

This should get solved once we move the Telemetry Service secrets out of the Maestro KV, I created a follow up issue for that

File: .vault-config/dotneteng-status-local.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    name: dotneteng-status-local
+
+references:
+  helixkv:
+    type: azure-key-vault
+    parameters:
+      subscription: a4fc5514-21a9-4296-bfaf-5c7ee7fa35d1
+      name: helixkv
+
+
```
Comment (riarenas) [inline .vault-config/dotneteng-status-local.yaml:14]: [nit] extra blank line

File: .vault-config/dotneteng-status-local.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+secrets:
+
```
Comment (riarenas) [inline .vault-config/dotneteng-status-local.yaml:16]: [nit] extra blank line

File: .vault-config/helixkv.yaml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+storageLocation:
+  type: azure-key-vault
+  parameters:
+    subscription: a4fc5514-21a9-4296-bfaf-5c7ee7fa35d1
+    name: helixkv
+
+
```
Comment (riarenas) [inline .vault-config/helixkv.yaml:7]: [nit] extra line

File: .vault-config/shared/dotnet-grafana-secrets.yaml [hunk 4]
```diff
@@ -0,0 +1,51 @@
+# API token from https://dotneteng-status{-staging}.azurewebsites.net/
+# Token should be named "dotent-grafana{-staging}/dotnet-build-bot-dotnet-eng-status-token"
```
Comment (riarenas) [inline .vault-config/shared/dotnet-grafana-secrets.yaml:2]:
```suggestion
# Token should be named "dotnet-grafana{-staging}/dotnet-build-bot-dotnet-eng-status-token"
```

**Review by riarenas — DISMISSED** [review]
I think this looks right to me. Didn't find anything obviously missing. Just a couple of file hygiene issues that are present in the arcade-services manifests and we shouldn't bring into here.

**Review by riarenas — APPROVED** [review]

**Pull request merged**